File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        HashSet<Character> set = new HashSet<>();

        int low = 0;
        int high = 0;
        int max = 0;
        int lngMax = 0;

        while (high < s.length()) {
            if (!set.contains(s.charAt(high))) {
                set.add(s.charAt(high));
                high++;
            } else {
                set.remove(s.charAt(low));
                low++;
            }
            max = high - low;
            lngMax = Math.max(max, lngMax);
        }

        return lngMax;
    }
}

File: 3-longest-substring-without-repeating-characters/longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {

        int i=0,j=0;
        unordered_map<char,int> mp;
        int n=s.length();
        int maxi=0,len=0;

        while(i<n){
            if(mp.find(s[i])==mp.end()){
             mp[s[i]]=i;
            len++;
            }
            else{
                j=max(mp[s[i]]+1,j);
                mp[s[i]]=i;
                len=i-j+1;
            }
           
            maxi=max(maxi,len);
           
            i++;
        }
       

        return maxi;


        
    }
};